packages/block-editor/src/components/block-list-appender/style.scss

```diff
@@ -31,6 +31,13 @@
 		transition: all 0.1s ease;
 		@include reduce-motion("transition");
 	}
+
+	// Show one appender at a time.
+	display: none;
+}
+
+.block-editor-block-list__block.is-selected .block-list-appender {
```

Comment:
If tests prove this is the behavior we want, this should be easy to enforce in JS (not render it at all) unless the block is selected.